### PR TITLE
Added integration tests for DCR API

### PR DESCRIPTION
**Pull request by Yasasr1**
Added integration tests for DCR API to incorporate following additional client metadata values

- aud
- software_id
- token_endpoint_auth_method
- id_token_encrypted_response_alg
- id_token_encrypted_response_enc

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java [hunk 1]
```diff
@@ -292,6 +337,153 @@ public void testRollbackOnInvalidRequest() throws IOException {
...
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC_VALUE);
+
+
+        StringEntity entity = new StringEntity(dcrRequest.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 400, "Service Provider " +
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:473]: Is this behaviour mandarated by the spec?
Or can the OP implementation decide this?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java [hunk 2]
```diff
@@ -117,6 +117,51 @@ public void testCreateServiceProviderRequest() throws IOException {
...
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
+
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+
+        Object responseObj = JSONValue.parse(rd);
+        EntityUtils.consume(response.getEntity());
+        client_id = ((JSONObject) responseObj).get("client_id").toString();
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:160]: Need to assert all optional attribute that was sent in the request to make sure they were persisted as expected

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java (hunk 1 above)
Comment (Yasasr1) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:473]: This is mandated by the spec 
[https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java (hunk 2 above)
Comment (Yasasr1) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:160]: added

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java [hunk 3]
```diff
@@ -117,6 +117,58 @@ public void testCreateServiceProviderRequest() throws IOException {
...
+        dcrRequest.put(OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC,
+                OAuthDCRMConstants.ID_TOKEN_ENCRYPTED_RESPONSE_ENC_VALUE);
+
+        StringEntity entity = new StringEntity(dcrRequest.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
```
Comment (dewniMW) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:154]: is the message provided here correct?

Comment (Yasasr1) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java:154]: This message will be displayed if the service provider is not created successfully. (if status code returned is not equal to 201)

**Comment by mefarazath** [comment]
Closed since the original feature PR is postponed

**Pull request closed without merging**